### PR TITLE
btop: update to 1.3.0, cleanup portfile

### PR DESCRIPTION
**Pull request by i0ntempest**
###### Type(s)

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS x.y
Xcode x.y / Command Line Tools x.y.z

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?
- [ ] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

File: sysutils/btop/Portfile [hunk 1]
```diff
@@ -3,46 +3,27 @@
...
 PortGroup           github 1.0
 PortGroup           makefile 1.0
-PortGroup compiler_blacklist_versions 1.0
+PortGroup           compiler_blacklist_versions 1.0
 
-github.setup        aristocratos btop 7e50b03e391a346b2b6ce92be332c58d91f75a00
-version             20230615
-
-fetch.type          git
+github.setup        aristocratos btop 1.3.0 v
```
Comment (reneeotten) [inline sysutils/btop/Portfile:8]: this version is likely not considered "newer" by MacPorts - please verify (i.e., is the port considered "outdated" if you have the previous version installed?). I doubt that is the case in which you should add "epoch 1".

**Review by reneeotten — CHANGES_REQUESTED** [review]

File: sysutils/btop/Portfile (hunk 1 above)
Comment (Samasaur1) [reply sysutils/btop/Portfile:8]: Can confirm that MacPorts does not consider this version newer without `epoch 1`, which does fix the problem

**Comment by Samasaur1** [comment]
We _should_ also [be able to use Apple's clang bundled with Xcode 15](https://github.com/aristocratos/btop/pull/666), since despite reporting a clang version of `Apple clang version 15.0.0 (clang-1500.1.0.2.5)`, it actually uses LLVM 16, which [is the version necessary for btop](https://github.com/aristocratos/btop/pull/510). Unfortunately, [the minimum versions of clang that MacPorts thinks is necessary for C++20](https://github.com/macports/macports-base/commit/f23dca5afb4bf4a01accb861d9095f9383e41930) does not support [ranges, the C++20 feature that btop needs](https://github.com/aristocratos/btop/issues/490), so we will need to maintain the list of blacklisted compilers.

**Comment by Samasaur1** [comment]
But that can be for a future improvement to the Portfile — once the `epoch 1` change is made, this PR looks ready to merge in my opinion

**Comment by i0ntempest** [comment]
Epoch addition has been pushed

**Pull request merged**